swiperefresh/src/main/java/com/google/accompanist/swiperefresh/SwipeRefresh.kt

```diff
@@ -114,10 +116,18 @@ class SwipeRefreshState(
     }
 }
 
-private class SwipeRefreshNestedScrollConnection(
-    private val state: SwipeRefreshState,
-    private val coroutineScope: CoroutineScope,
-    private val onRefresh: () -> Unit,
+/**
+ * The position for which the "swipe to refresh" must be enabled:
+ */
+enum class Position {
+    TOP,
+    BOTTOM,
+}
+
+private open class SwipeRefreshNestedScrollConnection(
```

Comment:
Can we remove the sub-class and just pass in a `scrollFromTop: Boolean = true`? Sub-classes are complicating the code here and there's no real need AFAICT.